src/gui/main_panel.cpp

```diff
@@ -448,48 +438,29 @@ MainPanel::init_loopers (int count)
 		while (count > (int) _looper_panels.size()) {
 			looperpan = new LooperPanel(this, _loop_control, _scroller, -1);
 			looperpan->set_index(_looper_panels.size());
-			_main_sizer->Add (looperpan, 0, wxEXPAND|wxALL, 0);
+			_scroller_sizer->Add (looperpan, 0, wxEXPAND|wxALL, 0);
 			_looper_panels.push_back (looperpan);
 		}
 	}
 	else if (count < (int)_looper_panels.size()) {
 		while (count < (int)_looper_panels.size()) {
 			looperpan = _looper_panels.back();
 			_looper_panels.pop_back();
-			_main_sizer->Remove((wxBoxSizer*)looperpan);
+			_scroller_sizer->Remove((wxBoxSizer*)looperpan);
 			looperpan->Destroy();
 		}
 	}
 
-	_scroller->SetClientSize(_scroller->GetClientSize());
 	_scroller->Layout();
-	_scroller->SetScrollRate(0,30);
-
- 	if (!_looper_panels.empty()) {
- 		wxSize bestsz = _looper_panels[0]->GetBestSize();
-		//cerr << "best w: " << bestsz.GetWidth() << endl;
- 		_scroller->SetMinClientSize (bestsz);
-		_topsizer->Layout();
-// 		_topsizer->Fit(this);
-// 		_topsizer->SetSizeHints(this);
-
-		
-		// maybe resize
-		if (_looper_panels.size() <= 4) {
-			int topheight = _top_panel->GetSize().GetHeight();
+	_scroller->FitInside();
 
-			//SetSize(GetSize().GetWidth(), bestsz.GetHeight() * _looper_panels.size()  + topheight); 
-			PreferredSizeChange(GetSize().GetWidth(), bestsz.GetHeight() * _looper_panels.size()  + topheight); // emit
-		}
-		
-		
+	// maybe resize topwindow, keeping width the same, but resize height to be just big enough to hold the updated number of looper panels
+	if (_looper_panels.size() > 0 && _looper_panels.size() <= 4) {
+		wxTopLevelWindow *topwindow = wxStaticCast(wxGetTopLevelParent(this), wxTopLevelWindow);
```

Comment:
hmm...maybe this should be dynamic cast like done in https://github.com/wxWidgets/wxWidgets/blob/cc3afe0dc60ddbd1ba8d31875e6f339fb55e1f93/src/univ/textctrl.cpp#L4807
`wxTopLevelWindow* topwindow = wxDynamicCast(wxGetTopLevelParent(this), wxTopLevelWindow);`